docs: add compiler for recursive Inline Methods (#15591)
lampepfl/dotty: docs/_docs/reference/metaprogramming/inline.md

Inline methods can be recursive. For instance, when called with a constant
exponent `n`, the following method for `power` will be implemented by

> straight inline code without any loop or recursion. It is worth to note that the number of successive

Reviewer: ```suggestion
straight inline code without any loop or recursion. It is worth noting that the number of successive
```

> inlines is limited and can be modified by the compiler setting `-Xmax-inlines`.

Reviewer: Maybe we should add what is the default value here ? Just to make it obvious as limitation leaves some uncertainty about that.

```suggestion
inlines is limited to 32 and can be modified by the compiler setting `-Xmax-inlines`.
```

> inlines is limited and can be modified by the compiler setting `-Xmax-inlines`.

Reviewer: Maybe we should add how nested it can be to make it obvious and not introduce uncertainty.

```suggestion
inlines is limited to 32 and can be modified by the compiler setting `-Xmax-inlines`.
```
